### DISCUSSION TITLE
Windows build fail

### DISCUSSION THREAD

**Discussion by J-Thierry** [rustdesk/rustdesk#1909] [Q&A] [answered]
hello 
I can't build last version of RustDesk.
the result is : 
------------------------------------------------------------------------------------------------------------------
D:\rustdesk>cargo run
   Compiling autocfg v1.1.0
   Compiling proc-macro2 v1.0.43
   Compiling unicode-ident v1.0.3
   Compiling quote v1.0.21
   Compiling cfg-if v1.0.0
   Compiling syn v1.0.99
   Compiling winapi v0.3.9
   Compiling memchr v2.5.0
   Compiling jobserver v0.1.24
   Compiling libc v0.2.132
   Compiling log v0.4.17
   Compiling once_cell v1.13.1
   Compiling serde_derive v1.0.144
   Compiling scopeguard v1.1.0
   Compiling serde v1.0.144
   Compiling lazy_static v1.4.0
   Compiling pin-project-lite v0.2.9
   Compiling futures-core v0.3.23
   Compiling smallvec v1.9.0
   Compiling windows_x86_64_msvc v0.36.1
   Compiling futures-task v0.3.23
   Compiling futures-sink v0.3.23
   Compiling futures-util v0.3.23
   Compiling pin-utils v0.1.0
   Compiling num_cpus v1.13.1
   Compiling hashbrown v0.12.3
   Compiling version_check v0.9.4
   Compiling parking_lot_core v0.9.3
   Compiling futures-io v0.3.23
   Compiling either v1.8.0
   Compiling regex-syntax v0.6.27
   Compiling rand_core v0.4.2
   Compiling bitflags v1.3.2
   Compiling humantime v2.1.0
   Compiling futures-channel v0.3.23
   Compiling itoa v1.0.3
   Compiling pkg-config v0.3.25
   Compiling anyhow v1.0.62
   Compiling fastrand v1.8.0
   Compiling protobuf v3.1.0
   Compiling unicode-width v0.1.9
   Compiling siphasher v0.2.3
   Compiling strsim v0.8.0
   Compiling vec_map v0.8.2
   Compiling glob v0.3.0
   Compiling zstd-safe v4.1.3+zstd.1.5.1
   Compiling signature v1.6.0
   Compiling ppv-lite86 v0.2.16
   Compiling winapi-build v0.1.1
   Compiling ryu v1.0.11
   Compiling heck v0.4.0
   Compiling crossbeam-utils v0.8.11
   Compiling rustversion v1.0.9
   Compiling dasp_sample v0.11.0
   Compiling minimal-lexical v0.2.1
   Compiling serde v0.9.15
   Compiling os_str_bytes v6.3.0
   Compiling dtoa v0.4.8
   Compiling itoa v0.3.4
   Compiling bindgen v0.59.2
   Compiling spin v0.5.2
   Compiling untrusted v0.7.1
   Compiling textwrap v0.15.0
   Compiling rustc-hash v1.1.0
   Compiling strsim v0.10.0
   Compiling shlex v1.1.0
   Compiling serde_json v1.0.85
   Compiling adler v1.0.2
   Compiling typenum v1.15.0
   Compiling peeking_take_while v0.1.2
   Compiling lazycell v1.3.0
   Compiling parking_lot_core v0.8.5
   Compiling camino v1.1.1
   Compiling winapi v0.2.8
   Compiling fnv v1.0.7
   Compiling unicode-segmentation v1.9.0
   Compiling semver v1.0.13
   Compiling rustls v0.20.6
   Compiling linked-hash-map v0.5.6
   Compiling cfg-if v0.1.10
   Compiling rayon-core v1.9.3
   Compiling windows_x86_64_msvc v0.28.0
   Compiling dasp_ring_buffer v0.11.0
   Compiling matches v0.1.9
   Compiling cbindgen v0.23.0
   Compiling gimli v0.26.2
   Compiling httparse v1.7.1
   Compiling tinyvec_macros v0.1.0
   Compiling windows_x86_64_msvc v0.30.0
   Compiling try-lock v0.2.3
   Compiling crc32fast v1.3.2
   Compiling rustc-demangle v0.1.21
   Compiling percent-encoding v2.1.0
   Compiling tfc v0.6.1 (https://github.com/asur4s/The-Fat-Controller#48303c5d)
   Compiling winit v0.26.1
   Compiling tower-service v0.3.2
   Compiling httpdate v1.0.2
   Compiling convert_case v0.5.0
   Compiling unicode-xid v0.2.3
   Compiling cty v0.2.2
   Compiling winreg v0.10.1
   Compiling pathdiff v0.2.1
   Compiling encoding_rs v0.8.31
   Compiling crossbeam-queue v0.3.6
   Compiling str-buf v1.0.6
   Compiling ntapi v0.3.7
   Compiling unicode-bidi v0.3.8
   Compiling iovec v0.1.4
   Compiling strum v0.24.1
   Compiling base64 v0.13.0
   Compiling sysinfo v0.24.7
   Compiling async-trait v0.1.57
   Compiling cpal v0.13.5
   Compiling time-macros v0.2.4
   Compiling widestring v1.0.2
   Compiling cpufeatures v0.2.4
   Compiling block v0.1.6
   Compiling ipnet v2.5.0
   Compiling widestring v0.4.3
   Compiling byteorder v1.4.3
   Compiling mime v0.3.16
   Compiling memalloc v0.1.0
   Compiling whoami v1.2.1
   Compiling getrandom v0.2.7
   Compiling instant v0.1.12
   Compiling autocfg v0.1.8
   Compiling cc v1.0.73
   Compiling slab v0.4.7
   Compiling lock_api v0.4.7
   Compiling indexmap v1.9.1
   Compiling tokio v1.21.1
   Compiling num-traits v0.2.15
   Compiling num-integer v0.1.45
   Compiling memoffset v0.6.5
   Compiling crossbeam-epoch v0.9.10
   Compiling rayon v1.5.3
   Compiling tracing-core v0.1.29
   Compiling rand_core v0.3.1
   Compiling proc-macro-error-attr v1.0.4
   Compiling proc-macro-error v1.0.4
   Compiling generic-array v0.14.6
   Compiling textwrap v0.11.0
   Compiling phf_shared v0.7.24
   Compiling ed25519 v1.5.2
   Compiling enigo v0.0.14 (D:\rustdesk\libs\enigo)
   Compiling kernel32-sys v0.2.2
   Compiling ws2_32-sys v0.2.1
   Compiling dasp_frame v0.11.0
   Compiling dasp_window v0.11.1
   Compiling clang-sys v1.3.3
   Compiling clap_lex v0.2.4
   Compiling miniz_oxide v0.5.3
   Compiling heck v0.3.3
   Compiling yaml-rust v0.4.5
   Compiling tinyvec v1.6.0
   Compiling form_urlencoded v1.0.1
   Compiling raw-window-handle v0.4.3
   Compiling rustls-pemfile v1.0.1
   Compiling addr2line v0.17.0
   Compiling rand_core v0.6.3
   Compiling uuid v1.1.2
   Compiling rand_chacha v0.1.1
   Compiling rand_pcg v0.1.2
   Compiling rand v0.6.5
   Compiling time v0.3.9
   Compiling zstd-sys v1.6.2+zstd.1.5.1
   Compiling ring v0.16.20
   Compiling backtrace v0.3.66
   Compiling webm-sys v1.0.3
   Compiling sys-locale v0.2.1
   Compiling virtual_display v0.1.0 (D:\rustdesk\libs\virtual_display)
   Compiling impersonate_system v0.1.0 (https://github.com/21pages/impersonate-system#af4a8205)
   Compiling clipboard v0.1.0 (D:\rustdesk\libs\clipboard)
   Compiling runas v0.2.1
   Compiling tracing v0.1.36
   Compiling rand_xorshift v0.1.1
   Compiling rand_hc v0.1.0
   Compiling rand_isaac v0.1.1
   Compiling phf v0.7.24
   Compiling dasp_rms v0.11.0
   Compiling dasp_peak v0.11.0
   Compiling dasp_interpolate v0.11.0
   Compiling dasp_slice v0.11.0
   Compiling unicode-normalization v0.1.21
   Compiling rand_chacha v0.3.1
   Compiling which v4.2.5
   Compiling error-code v2.3.1
   Compiling sciter-rs v0.5.57 (https://github.com/open-trade/rust-sciter?branch=dyn#82025b9b)
   Compiling want v0.3.0
   Compiling windows-sys v0.36.1
   Compiling aho-corasick v0.7.18
   Compiling nom v7.1.1
   Compiling object v0.29.0
   Compiling dasp_envelope v0.11.0
   Compiling crossbeam-channel v0.5.6
   Compiling windows-sys v0.28.0
   Compiling windows v0.30.0
   Compiling flate2 v1.0.24
   Compiling idna v0.2.3
   Compiling err-derive v0.3.1
   Compiling rand v0.8.5
   Compiling bytes v1.2.1
   Compiling toml v0.5.9
   Compiling serde_urlencoded v0.7.1
   Compiling winapi-util v0.1.5
   Compiling atty v0.2.14
   Compiling remove_dir_all v0.5.3
   Compiling rand_jitter v0.1.4
   Compiling rand_os v0.1.3
   Compiling libloading v0.7.3
   Compiling dirs-sys-next v0.1.2
   Compiling socket2 v0.4.6
   Compiling time v0.1.44
   Compiling winreg v0.6.2
   Compiling iana-time-zone v0.1.46
   Compiling mac_address v1.1.3
   Compiling socket2 v0.3.19
   Compiling num-traits v0.1.43
   Compiling net2 v0.2.37
   Compiling miow v0.3.7
   Compiling ansi_term v0.12.1
   Compiling clipboard-win v4.4.2
   Compiling system_shutdown v3.0.0
   Compiling errno v0.2.8
   Compiling ctrlc v3.2.3
   Compiling rpassword v7.0.0
   Compiling regex v1.6.0
   Compiling dasp_signal v0.11.0
   Compiling block-buffer v0.10.2
   Compiling crypto-common v0.1.6
   Compiling mio v0.8.4
   Compiling filetime v0.2.17
   Compiling repng v0.2.2
   Compiling cexpr v0.6.0
   Compiling url v2.2.2
   Compiling miow v0.4.0
   Compiling http v0.2.8
   Compiling same-file v1.0.6
   Compiling termcolor v1.1.3
   Compiling clap v2.34.0
   Compiling tempfile v3.3.0
   Compiling directories-next v2.0.0
   Compiling dirs-next v2.0.0
   Compiling machine-uid v0.2.0
   Compiling serde_json v0.9.10
   Compiling default-net v0.11.0
   Compiling crossbeam-deque v0.8.2
   Compiling chrono v0.4.22
   Compiling miow v0.2.2
   Compiling parking_lot v0.11.2
   Compiling dasp v0.11.0
   Compiling digest v0.10.5
   Compiling parking_lot v0.12.1
   Compiling walkdir v2.3.2
   Compiling http-body v0.4.5
   Compiling env_logger v0.9.0
   Compiling clap v3.2.17
   Compiling failure v0.1.8
   Compiling phf_generator v0.7.24
   Compiling synstructure v0.12.6
   Compiling mio v0.6.23
   Compiling sha2 v0.10.6
   Compiling which v3.1.1
   Compiling libsodium-sys v0.2.7
   Compiling phf_codegen v0.7.24
   Compiling mio-named-pipes v0.1.7
   Compiling trayicon v0.1.3-1 (https://github.com/open-trade/trayicon-rs#8d9c4489)
   Compiling target_build_utils v0.3.1
   Compiling thiserror-impl v1.0.32
   Compiling futures-macro v0.3.23
   Compiling tokio-macros v1.8.0
   Compiling pin-project-internal v1.0.12
   Compiling structopt-derive v0.4.18
   Compiling enum_dispatch v0.3.8
   Compiling enum-map-derive v0.10.0
   Compiling strum_macros v0.24.3
   Compiling clap_derive v3.2.17
   Compiling thiserror v1.0.32
   Compiling pin-project v1.0.12
   Compiling structopt v0.3.26
   Compiling enum-map v2.4.1
   Compiling windows-service v0.4.0
   Compiling protobuf-support v3.1.0
   Compiling confy v0.4.0 (https://github.com/open-trade/confy#630cc28a)
   Compiling arboard v2.1.1
   Compiling flexi_logger v0.22.6
   Compiling rdev v0.5.0-2 (https://github.com/asur4s/rdev#ea223720)
   Compiling sodiumoxide v0.2.7
   Compiling futures-executor v0.3.23
   Compiling cargo-platform v0.1.2
   Compiling serde_yaml v0.8.26
   Compiling futures v0.3.23
   Compiling tokio-util v0.7.4
   Compiling wol-rs v0.9.1
   Compiling scrap v0.5.0 (D:\rustdesk\libs\scrap)
   Compiling magnum-opus v0.4.0 (https://github.com/SoLongAndThanksForAllThePizza/magnum-opus#6247071a)
   Compiling winres v0.1.12
   Compiling cargo_metadata v0.14.2
   Compiling parity-tokio-ipc v0.7.3-1 (https://github.com/open-trade/parity-tokio-ipc#20b28959)
   Compiling tokio-socks v0.5.1-1 (https://github.com/open-trade/tokio-socks#7034e792)
   Compiling h2 v0.3.14
   Compiling protobuf-parse v3.1.0
   Compiling flutter_rust_bridge_codegen v1.32.0 (https://github.com/SoLongAndThanksForAllThePizza/flutter_rust_bridge#e5adce55)
   Compiling protobuf-codegen v3.1.0
   Compiling hyper v0.14.20
   Compiling webm v1.0.2
   Compiling hbb_common v0.1.0 (D:\rustdesk\libs\hbb_common)
   Compiling webpki v0.22.0
   Compiling sct v0.7.0
   Compiling webpki-roots v0.22.4
   Compiling zstd v0.9.2+zstd.1.5.1
   Compiling tokio-rustls v0.23.4
   Compiling hyper-rustls v0.23.0
   Compiling reqwest v0.11.11
   Compiling rustdesk v1.2.0 (D:\rustdesk)
warning: unused import: `crate::common::IS_X11`
 --> src\server\input_service.rs:2:5
  |
2 | use crate::common::IS_X11;
  |     ^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `TransferJobMeta`
  --> src\client\io_loop.rs:18:20
   |
18 |     RemoveJobMeta, TransferJobMeta,
   |                    ^^^^^^^^^^^^^^^

warning: unused import: `collections::HashMap`
 --> src\common.rs:2:5
  |
2 |     collections::HashMap,
  |     ^^^^^^^^^^^^^^^^^^^^

warning: unused import: `time`
  --> src\ui.rs:18:31
   |
18 |     tokio::{self, sync::mpsc, time},
   |                               ^^^^

warning: unused import: `crate::clipboard_file::*`
  --> src\ui\remote.rs:22:5
   |
22 | use crate::clipboard_file::*;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^

warning: unused import: `std::env::Args`
 --> src\core_main.rs:1:5
  |
1 | use std::env::Args;
  |     ^^^^^^^^^^^^^^

warning: unused import: `empty_clipboard`
  --> src\ui_cm_interface.rs:14:48
   |
14 | use clipboard::{cliprdr::CliprdrClientContext, empty_clipboard, ContextSend};
   |                                                ^^^^^^^^^^^^^^^

warning: unreachable expression
   --> src\ui_interface.rs:348:5
    |
346 |         return crate::platform::is_installed();
    |         -------------------------------------- any code following this expression is unreachable
347 |     }
348 |     false
    |     ^^^^^ unreachable expression
    |
    = note: `#[warn(unreachable_code)]` on by default

warning: unreachable expression
   --> src\ui_interface.rs:801:5
    |
800 |     return crate::platform::is_root();
    |     --------------------------------- any code following this expression is unreachable
801 |     false
    |     ^^^^^ unreachable expression

warning: unreachable expression
    --> src\ui_session_interface.rs:1318:25
     |
1310 |                             return None;
     |                             ----------- any code following this expression is unreachable
...
1318 |                         None
     |                         ^^^^ unreachable expression

warning: unused import: `FileManager`
 --> src\client\io_loop.rs:2:26
  |
2 |     Client, CodecFormat, FileManager, MediaData, MediaSender, QualityStatus, MILLI1, SEC30,
  |                          ^^^^^^^^^^^

warning: value assigned to `disable_numlock` is never read
   --> src\server\input_service.rs:728:13
    |
728 |     let mut disable_numlock = false;
    |             ^^^^^^^^^^^^^^^
    |
    = note: `#[warn(unused_assignments)]` on by default
    = help: maybe it is overwritten before being read?

warning: variable `would_block_count` is assigned to, but never used
   --> src\server\video_service.rs:471:13
    |
471 |     let mut would_block_count = 0u32;
    |             ^^^^^^^^^^^^^^^^^
    |
    = note: `#[warn(unused_variables)]` on by default
    = note: consider using `_would_block_count` instead

warning: function `disable_keyboard_listening` is never used
    --> src\client.rs:1950:8
     |
1950 | pub fn disable_keyboard_listening() {
     |        ^^^^^^^^^^^^^^^^^^^^^^^^^^
     |
     = note: `#[warn(dead_code)]` on by default

warning: type alias `Status` is never used
  --> src\ui.rs:37:6
   |
37 | type Status = (i32, bool, i64, String);
   |      ^^^^^^

warning: function `check_connect_status` is never used
  --> src\ui.rs:46:4
   |
46 | fn check_connect_status(
   |    ^^^^^^^^^^^^^^^^^^^^

warning: function `get_clients_state` is never used
   --> src\ui_cm_interface.rs:222:8
    |
222 | pub fn get_clients_state() -> String {
    |        ^^^^^^^^^^^^^^^^^

warning: type alias `Message` is never used
  --> src\ui_interface.rs:26:6
   |
26 | type Message = RendezvousMessage;
   |      ^^^^^^^

warning: function `get_id` is never used
  --> src\ui_interface.rs:56:8
   |
56 | pub fn get_id() -> String {
   |        ^^^^^^

warning: function `get_sound_inputs` is never used
   --> src\ui_interface.rs:228:8
    |
228 | pub fn get_sound_inputs() -> Vec<String> {
    |        ^^^^^^^^^^^^^^^^

warning: function `get_app_name` is never used
   --> src\ui_interface.rs:610:8
    |
610 | pub fn get_app_name() -> String {
    |        ^^^^^^^^^^^^

warning: function `discover` is never used
   --> src\ui_interface.rs:648:8
    |
648 | pub fn discover() {
    |        ^^^^^^^^

warning: function `change_id` is never used
   --> src\ui_interface.rs:692:8
    |
692 | pub fn change_id(id: String) {
    |        ^^^^^^^^^

warning: function `check_super_user_permission` is never used
   --> src\ui_interface.rs:805:8
    |
805 | pub fn check_super_user_permission() -> bool {
    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: function `send_to_cm` is never used
   --> src\ui_interface.rs:909:21
    |
909 | pub(crate) async fn send_to_cm(data: &ipc::Data) {
    |                     ^^^^^^^^^^

warning: constant `INVALID_FORMAT` is never used
   --> src\ui_interface.rs:915:7
    |
915 | const INVALID_FORMAT: &'static str = "Invalid format";
    |       ^^^^^^^^^^^^^^

warning: constant `UNKNOWN_ERROR` is never used
   --> src\ui_interface.rs:916:7
    |
916 | const UNKNOWN_ERROR: &'static str = "Unknown error";
    |       ^^^^^^^^^^^^^

warning: function `change_id_` is never used
   --> src\ui_interface.rs:919:10
    |
919 | async fn change_id_(id: String, old_id: String) -> &'static str {
    |          ^^^^^^^^^^

warning: function `check_id` is never used
   --> src\ui_interface.rs:967:10
    |
967 | async fn check_id(
    |          ^^^^^^^^

warning: `rustdesk` (lib) generated 29 warnings
    Finished dev [unoptimized + debuginfo] target(s) in 7m 34s
warning: the following packages contain code that will be rejected by a future version of Rust: ntapi v0.3.7, winapi v0.2.8
note: to see what the problems were, use the option `--future-incompat-report`, or run `cargo report future-incompatibilities --id 1`
     Running `target\debug\rustdesk.exe`
error: process didn't exit successfully: `target\debug\rustdesk.exe` (exit code: 0xc0000139, STATUS_ENTRYPOINT_NOT_FOUND)
------------------------------------------------------------------------------------------------------------------
when I see "Running `target\debug\rustdesk.exe`" I have this error message in popup warning :
**_rustdesk.exe - Entry point not found
Procedure entry point DiUninstallDriverW not found in dynamic link library
\\?D:\rustdesk\target\debug\rustdesk.exe_**

Can you help me for solve this build problem ?
Excuse me for my English, but I hope I was clear.
best regard

**Comment by djex** [comment]
Are you running on Windows 7?

**Comment by J-Thierry** [comment]
Hello
No.
I use Windows server 2016 version 1607

**Reply by mrwrighty** [reply]
Same issue Windows 10 same errors as above. cargo run creates and runs debug version OK

**Comment by tahssinaldrajee** [comment]
try to repeat [this](https://github.com/rustdesk/rustdesk#:~:text=alsa%2Dlib%20pipewire-,Install%20vcpkg,-git%20clone%20https) step without checkout git and install libvpx libyuv opus

**Reply by djex** [reply]
This is what I had to do to get my vcpkg to install. The older checkout version suggested has some broken dependency urls.

**Comment by devme19** [comment]
all you need to get windows build
after isntall all dependencies in https://rustdesk.com/docs/en/dev/build/windows/ 
Generating flutter rust bridge files (rustdesk/flutter/run.sh)
rename python3 with python in build.py then
Running python build.py --flutter

**Comment by rustdesk** [comment, marked as answer]
https://discord.com/invite/nDceKgxnkV build channel
